src/executor/seq_scan_executor.py

```diff
@@ -51,4 +51,5 @@ def exec(self) -> Iterator[Batch]:
                 batches = [expr.evaluate(batch) for expr in self.project_expr]
                 batch = Batch.merge_column_wise(batches)
 
-            yield batch
+            if not batch.empty():
```

Comment:
Do we need this check? Yielding an empty batch might make things easy for the caller

Reply:
I move this check to when we apply predicate, and it also works.